components/omnibox/browser/suggested_sites_provider_data.cc

```diff
@@ -3,98 +3,147 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "brave/components/binance/browser/regions.h"
+#include "brave/components/ntp_widget_utils/browser/ntp_widget_utils_region.h"
 #include "brave/components/omnibox/browser/suggested_sites_provider.h"
 
 #include "base/strings/utf_string_conversions.h"
 
+namespace {
+
+const char binance_host[] = "binance.com";
+const char binance_host_full[] = "wwww.binance.com";
+const char buy_sell_path[] = "/buy-sell-crypto/";
+
+std::string GetLocalizedURL(const std::string& query_params, bool str_display) {
+  const std::string url_locale =
+      ntp_widget_utils::FindLocale(::binance::supported_locales, "en");
+  const std::string path = "/" + url_locale + std::string(buy_sell_path);
+  const std::string protocol =
+      !str_display ? (std::string(url::kHttpsScheme) + "://") : std::string();
+  const std::string host = str_display ? binance_host : binance_host_full;
+  const std::string url = protocol + host + path + query_params;
```

Comment:
A bit cleaner to use `base::StringPrintf`

Reply:
Cleaned up this function